benchmarking/src/cloudflare.ts

```diff
@@ -0,0 +1,95 @@
+import nodeFsPromises from "node:fs/promises";
+import nodeFs from "node:fs";
+import nodePath from "node:path";
+import nodeChildProcess from "node:child_process";
+import nodeUtil from "node:util";
+
+const promiseExec = nodeUtil.promisify(nodeChildProcess.exec);
+
+await ensureWranglerSetup();
+
+/**
+ * Collects name and absolute paths of apps (in this repository) that we want to benchmark
+ *
+ * @returns Array of objects containing the app's name and absolute path
+ */
+export async function collectAppPathsToBenchmark(): Promise<
+  {
+    name: string;
+    path: string;
+  }[]
+> {
+  const allExampleNames = await nodeFsPromises.readdir("../examples");
+
+  const examplesToIgnore = new Set(["vercel-commerce"]);
+
+  const examplePaths = allExampleNames
+    .filter((exampleName) => !examplesToIgnore.has(exampleName))
+    .map((exampleName) => ({
+      name: exampleName,
+      path: nodePath.resolve(`../examples/${exampleName}`),
+    }));
+
+  return examplePaths;
+}
+
+/**
+ * Builds an application using their "build:worker" script
+ * (an error is thrown if the application doesn't have such a script)
+ *
+ * @param dir Path to the application to build
+ */
+export async function buildApp(dir: string): Promise<void> {
+  const packageJsonPath = `${dir}/package.json`;
+  if (!nodeFs.existsSync(packageJsonPath)) {
+    throw new Error(`Error: package.json for app at "${dir}" not found`);
+  }
+
+  const packageJsonContent = JSON.parse(await nodeFsPromises.readFile(packageJsonPath, "utf8"));
+
+  if (!("scripts" in packageJsonContent) || !("build:worker" in packageJsonContent.scripts)) {
+    throw new Error(`Error: package.json for app at "${dir}" does not include a "build:worker" script`);
+  }
+
+  const command = "pnpm build:worker";
+
+  await promiseExec(command, { cwd: dir });
+}
+
+/**
+ * Deploys a built application using wrangler
+ *
+ * @param dir Path to the application to build
+ * @returns the url of the deployed application
+ */
+export async function deployBuiltApp(dir: string): Promise<string> {
+  const { stdout } = await promiseExec("pnpm exec wrangler deploy", { cwd: dir });
```

Comment:
I've used the promisified version of `exec` so that we can get a nice clean `stdout` here (as mentioned in the [nodejs docs](https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback))

I think that extracting the `stdout` from the standard sync `exec` would be much longer and uglier than this`promiseExec` solution

Reply:
I'm not sure to get your point here. Isn't stdout passed to the sync callback?

Reply:
yes, I can for sure use the sync version of `exec` I was just saying that I think that that's going to make the code longer and uglier than the current solution that promisifies `exec`

But if you really dislike the promisified version of `exec` and prefer the sync version I don't mind switching to that (personally I find it worse but I'm fine with either)

Reply:
I guess it boils down to code style.

I find code easier to grasp when:
- it depends on "local" inputs (`promiseExec` is defined 60l up)
- it doesn't depend on global vars
- names are explicit - i.e. `promisifiedExec`

But it works the same, pick the version you're most comfortable with.

Reply:
I've removed `promiseExec`: https://github.com/opennextjs/opennextjs-cloudflare/pull/117/commits/62e878b6b01b2704e74a3d83957e0a65291a8547

Please let me know if you're happy with the changed version 🙂